homeassistant/components/homeassistant/triggers/event.py

```diff
@@ -19,36 +20,42 @@
         vol.Required(CONF_PLATFORM): "event",
         vol.Required(CONF_EVENT_TYPE): cv.string,
         vol.Optional(CONF_EVENT_DATA): dict,
+        vol.Optional(CONF_EVENT_CONTEXT): dict,
     }
 )
 
 
+def _populate_schema(config, config_parameter):
+    if config_parameter not in config:
+        return None
+
+    return vol.Schema(
+        {vol.Required(key): value for key, value in config[config_parameter].items()},
+        extra=vol.ALLOW_EXTRA,
+    )
+
+
 async def async_attach_trigger(
     hass, config, action, automation_info, *, platform_type="event"
 ):
     """Listen for events based on configuration."""
     event_type = config.get(CONF_EVENT_TYPE)
-    event_data_schema = None
-    if config.get(CONF_EVENT_DATA):
-        event_data_schema = vol.Schema(
-            {
-                vol.Required(key): value
-                for key, value in config.get(CONF_EVENT_DATA).items()
-            },
-            extra=vol.ALLOW_EXTRA,
-        )
+    event_data_schema = _populate_schema(config, CONF_EVENT_DATA)
+    event_context_schema = _populate_schema(config, CONF_EVENT_CONTEXT)
```

Comment:
Just realized that matching straight up on context like this is going to be annoying if one wants to match multiple users.

Reply:
Good point. How about something like:
```python
if isinstance(value, list):
    vol.Required(key): vol.In(value)
else:
    vol.Required(key): value
```
when building the schema? This can also be used for `event_data`, not just context.

Reply:
Downside is that this could be a breaking change if the user intended to match a list.

Reply:
I changed it here:
https://github.com/home-assistant/core/pull/41036

To ensure backwards compatibility I only added this for context, not data, even though I think it could be a nice touch for data as well.